test/e2e/kappcontroller/fetch_test.go

```diff
@@ -0,0 +1,168 @@
+	sas := e2e.ServiceAccounts{env.Namespace}
+	name := "test-app-fetch-dirname-single"
+
+	appSingleFetchYAML := `---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: test-app-fetch-dirname-single
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  fetch:
+  - inline:
+      destDirName: foo
+      paths:
+        file.csv: |
+            Year,Make,Model,Description,Price
+            1997,Ford,E350,"ac, abs, moon",3000.00
+  template:
+  - ytt:
+      paths:
+      - foo
+      inline:
+        paths:
+          file.yml: |
+            #@ load("@ytt:data", "data")
+            apiVersion: v1
+            kind: ConfigMap
+            metadata:
+              name: configmap
+            data:
+              value: #@ data.read("file.csv")
+  deploy:
+  - kapp: {}
+` + sas.ForNamespaceYAML()
+
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("deploy", func() {
+		// if template stage succeeds, assume test pass
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name},
+			e2e.RunOpts{StdinReader: strings.NewReader(appSingleFetchYAML)})
+	})
+
+	out := kapp.Run([]string{"inspect", "-a", name, "--raw", "--tty=false", "--filter-kind=App"})
+
+	var cr v1alpha1.App
+	err := yaml.Unmarshal([]byte(out), &cr)
+	if err != nil {
+		t.Fatalf("Failed to unmarshal: %s", err)
+	}
+
+	if !strings.Contains(cr.Status.Fetch.Stdout, "path: foo") {
+		t.Fatalf("Expected destDirName to be used by vendir")
+	}
+}
+
+func Test_App_FetchDirname_All(t *testing.T) {
+	env := e2e.BuildEnv(t)
+	logger := e2e.Logger{}
+	kapp := e2e.Kapp{t, env.Namespace, logger}
+	sas := e2e.ServiceAccounts{env.Namespace}
+	name := "test-app-fetch-dirname-all"
+
+	appAllFetchYAML := `---
+apiVersion: kappctrl.k14s.io/v1alpha1
+kind: App
+metadata:
+  name: test-app-fetch-dirname-multiple
+  annotations:
+    kapp.k14s.io/change-group: kappctrl-e2e.k14s.io/apps
+spec:
+  serviceAccountName: kappctrl-e2e-ns-sa
+  fetch:
+  - inline:
+      destDirName: foo
+      paths:
+        file.csv: |
+            Year,Make,Model,Description,Price
+            1997,Ford,E350,"ac, abs, moon",3000.00
+  - git:
+      url: https://github.com/k14s/kapp
+      ref: origin/develop
+      subPath: examples/gitops/guestbook
+      destDirName: git-dest
+  - http:
+      url: https://raw.githubusercontent.com/k14s/kapp/db2cc63e12e988235eb8815af8edc0ca0cfaa79c/examples/simple-app-example/config-1.yml
+      sha256: 71b0a2dceb6bb6e7a519d0587abd9400a265cabbe2c084f783df94a92db6d980
+      destDirName: http-dest
+  - imgpkgBundle:
+      image: k8slt/kctrl-example-pkg:v1.0.0
+      destDirName: imgpkg-dest
+  - helmChart:
+      name: redis
+      # Chart version v1, DEPRECATED 
+      version: "11.3.4"
+      repository:
+        url: https://charts.bitnami.com/bitnami
+      destDirName: helm-dest
+  template:
+  - ytt:
+      paths:
+      - foo
```

Comment:
i'm trying to imagine having a meaningful test of each of the fetches above.

i guess what if we had a series of ytt templating steps, each of which basically only included one of the named paths and then just put the contents (or part of the contents) of a specific named file from the fetched-contents into the body of a configmap?   that's only a little bit tedious to set up?

Reply:
That would the fetched contents look like for something like helmTemplate? I did the check for the path being used by vendir for those cases where we don't really have a thing for ytt to read. Or at least not a common thing, I guess we could hardcode the contents of each test artifact?

Like if we know git repo has a file foo.yml in it we use that?

Reply:
> Like if we know git repo has a file foo.yml in it we use that?

That's what I was thinking.. ?   doesn't need to be that way.  You're right that those vendir assertions are a sort of minimal coverage,  maybe that's enough then.

Reply:
I added the tests, turns out using 
```
  template:
  - ytt:
      paths:
      - foo
```
tells ytt to evaluate the whole directory, so adding the configmap was kind of redundant. As ytt will fail if that directory doesn't exist.